rust/lance-table/src/io/commit/external_manifest.rs

```diff
@@ -43,6 +43,20 @@ pub trait ExternalManifestStore: std::fmt::Debug + Send + Sync {
     /// the version and the store should not customize it.
     async fn get_latest_version(&self, base_uri: &str) -> Result<Option<(u64, String)>>;
 
+    /// Get the latest manifest location for a given base_uri.
+    async fn get_latest_manifest_location(
```

Comment:
Should we comment this so it is clear that users can/should override it if they are able to store the size in the external store? (e.g. in dynamodb we should be able to add a size column)

```suggestion
    /// Get the latest manifest location for a given base_uri.
    ///
    /// By default, this calls get_latest_version.  Impls should
    /// override this method if they store both the location and size
    /// of the latest manifest.
    async fn get_latest_manifest_location(
```